pkg/blackduck/ctl_blackduck.go

```diff
@@ -301,11 +287,23 @@ func (ctl *Ctl) SetFlag(f *pflag.Flag) {
 			ctl.Spec.ScanType = ctl.ScanType
 		case "persistent-storage":
 			ctl.Spec.PersistentStorage = ctl.PersistentStorage
-		case "pvc":
-			for _, pvcJSON := range ctl.PVCJSONSlice {
-				pvc := &blackduckv1.PVC{}
-				json.Unmarshal([]byte(pvcJSON), pvc)
-				ctl.Spec.PVC = append(ctl.Spec.PVC, *pvc)
+		case "pvc-file":
+			data, err := util.ReadFileData(ctl.PVCFile)
+			if err != nil {
+				log.Errorf("failed to read pvc file: %s", err)
+			}
+			type PVCStructs struct {
+				Data []blackduckv1.PVC
```

Comment:
A bit confused about this, why is the struct defined here?

Reply:
In Go you have to tell it the format of the data before you can convert it. The format of the input file will look like... 
```
{
  "Data" : [
    {
      "name": "bob",
      "size": "20"
    },
    {
      "name": "joe",
      "size": "30"
    }
  ]
}
```

Reply:
But can we move it out of the function, because it seems every time a caller steps through the control logic it will create a new same definition, right?